types/opt/bool.go

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package opt defines optional types.
+package opt
+
+import "strconv"
+
+// Bool represents an optional boolean to be JSON-encoded.
+// The string can be empty (for unknown or unspecified), or
+// "true" or "false".
+type Bool string
```

Comment:
I'm curious if this is compatible with Swift's JSON decoder. By default, an empty opt.Bool in a struct is going to be encoded by Go as `{"B":""}`. Can Swift decode that with a `: Bool` or a `: Bool`?

I see why you're doing this (the type itself cannot carry the `omitempty` attribute) but it's unfortunate we have to use an empty string to omit a value in JSON, when JSON has a nicer way of representing this (use real boolean types and omit the field). I'd prefer a real ```MyValue bool `json:"my_value,omitempty"` ``` if we can.

Reply:
Fixed in new separate PR.